vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs

```diff
@@ -189,33 +193,51 @@ type internal FSharpCompletionProvider
                         | CompletionItemKind.Argument -> 4
                         | CompletionItemKind.Other -> 5
                         | CompletionItemKind.Method (isExtension = true) -> 6
-                    kindPriority, not item.IsOwnMember, item.Name.ToLowerInvariant(), item.MinorPriority)
+                    not item.IsResolvable, kindPriority, not item.IsOwnMember, item.Name.ToLowerInvariant(), item.MinorPriority)
 
             let maxHints = if mruItems.Values.Count = 0 then 0 else Seq.max mruItems.Values
 
-            sortedDeclItems |> Array.iteri (fun number declarationItem ->
-                let glyph = CommonRoslynHelpers.FSharpGlyphToRoslynGlyph (declarationItem.Glyph, declarationItem.Accessibility)
-                let name =
+            sortedDeclItems |> Array.iteri (fun number declItem ->
+                let glyph = CommonRoslynHelpers.FSharpGlyphToRoslynGlyph (declItem.Glyph, declItem.Accessibility)
+                let name, namespaceToOpen =
                     match entityKind with
-                    | Some EntityKind.Attribute when declarationItem.IsAttribute && declarationItem.Name.EndsWith "Attribute"  ->
-                        declarationItem.Name.[0..declarationItem.Name.Length - attributeSuffixLength - 1] 
-                    | _ -> declarationItem.Name
+                    | Some EntityKind.Attribute when declItem.IsAttribute && declItem.Name.EndsWith "Attribute"  ->
+                        declItem.Name.[0..declItem.Name.Length - attributeSuffixLength - 1], None
+                    | _ when not declItem.IsResolvable ->
+                        let ns =
+                            match declItem.FullName.LastIndexOf '.' with
+                            | -1 -> ""
```

Comment:
What happens if namespace is empty? We then suffix '(open )' to the sprintf. 

Also does this definitely not attempt to open static classes etc? How accurate is this heuristic?